managed/src/main/java/com/yugabyte/yw/commissioner/tasks/UniverseTaskBase.java

```diff
@@ -316,6 +316,9 @@ public SubTaskGroup createDestroyServerTasks(Collection<NodeDetails> nodes,
       if (node.cloudInfo.private_ip == null) {
         LOG.warn(String.format("Node %s doesn't have a private IP. Skipping node delete.",
                                node.nodeName));
+        // Free up the node so that the client can use the instance to create another universe.
```

Comment:
`NodeInstance` objects only exist for on-prem providers. We can't always run this command. We need to verify if the provider is an on-premise one or not, and if so, then only clear the state of the node instances.